qcodes/instrument_drivers/QDevil/QDevil_QDac_channels.py

```diff
@@ -0,0 +1,833 @@
+
+        # isQDevilQDAC  = self._check_if_qdevilqdac()
+        firmware_version = self._get_firmware_version()
+        if firmware_version < 1.07:
+            log.warning("Firmware version: {}".format(firmware_version))
+            raise RuntimeError( '''
+                                No QDevil QDAC detected or the firmware version is obsolete.
+                                QCoDeS only supports version 1.07 or newer. In that case, 
+                                please Contact info@qdevil.com for an update.
+                                ''')
+
+        self.num_chans = self._get_number_of_channels()
+        num_boards = int(self.num_chans/8)
+        self._output_n_lines = self.num_chans + 2
+
+        # Assigned slopes. Entries will eventually be [chan, slope]
+        self._slopes = []
+        # Function generators (used in _set_voltage)
+        self._fgs = set(range(1, 9))
+        self._assigned_fgs = {}  # {chan: fg}
+        # Sync channels
+        self._syncoutputs = []  # Entries: [chan, syncchannel]
+
+        self.chan_range = range(1, 1 + self.num_chans)
+        self.channel_validator = vals.Ints(1, self.num_chans)
+
+        channels = ChannelList(self, "Channels", QDacChannel,
+                               snapshotable=False,
+                               multichan_paramclass=QDacMultiChannelParameter)
+
+        for i in self.chan_range:
+            channel = QDacChannel(self, 'chan{:02}'.format(i), i)
+            channels.append(channel)
+            # Should raise valueerror if name is invalid (silently fails now)
+            self.add_submodule('ch{:02}'.format(i), channel)
+        channels.lock()
+        self.add_submodule('channels', channels)
+
+        for board in range(num_boards):
+            for sensor in range(3):
+                label = 'Board {}, Temperature {}'.format(board, sensor)
+                self.add_parameter(name='temp{}_{}'.format(board, sensor),
+                                   label=label,
+                                   unit='C',
+                                   get_cmd='tem {} {}'.format(board, sensor),
+                                   get_parser=self._num_verbose)
+
+        self.add_parameter(name='cal',
+                           set_cmd='cal {}',
+                           vals=self.channel_validator)
+        # TO-DO: maybe it's too dangerous to have this settable.
+        # And perhaps ON is a better verbose mode default?
+        self.add_parameter(name='verbose',
+                           set_cmd='ver {}',
+                           val_mapping={True: 1, False: 0})
+
+        # Initialise the instrument, all channels DC (unbind func. generators)
+        for chan in self.chan_range:
+            # Note: this call does NOT change the voltage on the channel
+            self.write('wav {} 0 1 0'.format(chan))
+
+        # Get all calibrated min/max output values, before swithcing to verbose off mode
+        self.vranges = {}
```

Comment:
Very good points. Therefore I now refactored and invented a new parameter "mode" which controls both voltage and current ranges under the hood. 
Now, if the user needs to know the actual allowable voltage range for a channel it can be read from "ch##.v.vals", which is always up to date.